### ISSUE TITLE
[Helix] WebApplicationFactory test needs additional publish files to work on helix

### ISSUE THREAD

**Issue by HaoK** [dotnet/aspnetcore#7986]
Currently all of these kinds of tests are disabled on helix because they need to include additional things for these style tests to work

Helix tests are basically just doing a dotnet publish, copying the publish folder over as is, and run via dotnet vstest on helix machines.  

These kinds of tests appear to need extra things like deps files, as well as potentially things like related views assemblies, for example:

```
 System.IO.FileNotFoundException : Related assembly 'Microsoft.AspNetCore.Identity.UI.Views.V3' specified by assembly 'Microsoft.AspNetCore.Identity.UI' could not be found in the directory C:\temp\identity. Related assemblies must be co-located with the specifying assemblies.
Stack Trace:
   at Microsoft.AspNetCore.Mvc.ApplicationParts.RelatedAssemblyAttribute.GetRelatedAssemblies(Assembly assembly, Boolean throwOnError, Func`2 fileExists, Func`2 loadFile) in C:\Github\AspNetCore\src\Mvc\Mvc.Core\src\ApplicationParts\RelatedAssemblyAttribute.cs:line 104
   at Microsoft.AspNetCore.Mvc.ApplicationParts.RelatedAssemblyAttribute.GetRelatedAssemblies(Assembly assembly, Boolean throwOnError) in C:\Github\AspNetCore\src\Mvc\Mvc.Core\src\ApplicationParts\RelatedAssemblyAttribute.cs:line 53
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.<>c__DisplayClass3_0.<AddRelatedParts>b__0(ApplicationPartManager partManager) in C:\Github\AspNetCore\src\Identity\UI\src\IdentityBuilderUIExtensions.cs:line 79
   at Microsoft.Extensions.DependencyInjection.MvcCoreMvcBuilderExtensions.ConfigureApplicationPartManager(IMvcBuilder builder, Action`1 setupAction) in C:\Github\AspNetCore\src\Mvc\Mvc.Core\src\DependencyInjection\MvcCoreMvcBuilderExtensions.cs:line 108
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddRelatedParts(IdentityBuilder builder, UIFramework framework) in C:\Github\AspNetCore\src\Identity\UI\src\IdentityBuilderUIExtensions.cs:line 74
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI(IdentityBuilder builder, UIFramework framework) in C:\Github\AspNetCore\src\Identity\UI\src\IdentityBuilderUIExtensions.cs:line 53
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI(IdentityBuilder builder) in C:\Github\AspNetCore\src\Identity\UI\src\IdentityBuilderUIExtensions.cs:line 33
   at Microsoft.Extensions.DependencyInjection.IdentityServiceCollectionUIExtensions.AddDefaultIdentity[TUser](IServiceCollection services, Action`1 configureOptions) in C:\Github\AspNetCore\src\Identity\UI\src\IdentityServiceCollectionUIExtensions.cs:line 56
```

Basically what we need is a self contained directory that has everything needed to be able to run the tests after publish.  Hopefully this is just copying a few additional files from the bin directory?

@mkArtakMSFT @javiercn @pranavkm

**Comment by HaoK** [comment]
Projects affected by this include:

```
src\Azure\AzureAD\test\FunctionalTests\Microsoft.AspNetCore.Authentication.AzureAD.FunctionalTests.csproj
src\Identity\test\Identity.FunctionalTests\Microsoft.AspNetCore.Identity.FunctionalTests.csproj
src\Mvc\test\Mvc.FunctionalTests\Microsoft.AspNetCore.Mvc.FunctionalTests.csproj
```

**Comment by javiercn** [comment]
Functional tests are not designed to work against published content. There is extensibility to enable this scenario:
* You'll have to define some sort of layout for the wwwroots of the targeted projects in the published output. For example:
  * publish\Websites\BasicWebSite\wwwroot
  * publish\Websites\FilesWebSite\wwwroot
* You'll have to define assembly level attributes WebApplicationFactoryContentRootAttribute for all the attributes.
  * WebApplicationFactoryContentRootAttribute supports relative paths (I believe from the executing assembly).
* You'll have to copy the deps file for each of the websites into the published content.

I suggest adding an AssemblyInfo.Helix.cs to Functionaltests for defining the attributes and a FunctionalTests.Helix.targets file for defining the MSBuild for doing the file copy.

After that the tests should work provided that shadow copying is disabled.

**Comment by HaoK** [comment]
Addressed by https://github.com/dotnet/aspnetcore/issues/27296
